Wikipedia/Code/WMFWelcomeAnimationViewControllers.swift

```diff
@@ -0,0 +1,85 @@
+class WMFWelcomeAnimationViewController: UIViewController {
+    var welcomePageType:WMFWelcomePageType = .intro
+    fileprivate var hasAlreadyAnimated = false
+    
+    open lazy var animationView: WMFWelcomeAnimationView? = nil
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        guard let animationView = animationView else {return}
+        animationView.translatesAutoresizingMaskIntoConstraints = false
+        view.wmf_addSubviewWithConstraintsToEdges(animationView)
+    }
+    
+    override func didMove(toParentViewController parent: UIViewController?) {
+        super.didMove(toParentViewController: parent)
+        view.superview?.layoutIfNeeded() // Fix for: http://stackoverflow.com/a/39614714
```

Comment:
Seems like this should be fixed in the parent view controller rather than having this VC reach outside of itself to tell a superview to layout.

Reply:
I have the fix for this if you'd rather me just commit it, let me know. The custom view can handle all of this in `layoutSubviews`. You could also adjust the corner radius property there.

Reply:
@joewalsh 

Turns out this was needed to fix the issue online in the comment's stack overflow link.

You can see the difference on iPhone 5 - we tweak constraints in `reduceTopAnimationsSizes`.

With it the animations appear with reduced size:
<img width="376" alt="screen shot 2017-10-02 at 4 35 52 pm" src="https://user-images.githubusercontent.com/3143487/31104221-9ab5fca0-a790-11e7-97d1-6fe655fcfc18.png">

Without it they don't:
<img width="376" alt="screen shot 2017-10-02 at 4 37 56 pm" src="https://user-images.githubusercontent.com/3143487/31104234-ab5602b2-a790-11e7-84ac-7a176c1c4e17.png">

Reply:
iirc this started happening in iOS8... not sure if there's a better work-around - open to ideas... but this fix seems to have been working ok

Reply:
i think it's specific to using container views...